static/app/gettingStartedDocs/javascript/react-router.tsx

```diff
@@ -0,0 +1,507 @@
+  let message = "Oops!";
+  let details = "An unexpected error occurred.";
+  let stack: string | undefined;
+
+  if (isRouteErrorResponse(error)) {
+    message = error.status === 404 ? "404" : "Error";
+    details =
+      error.status === 404
+        ? "The requested page could not be found."
+        : error.statusText || details;
+  } else if (error && error instanceof Error) {
+    // you only want to capture non 404-errors that reach the boundary
+    Sentry.captureException(error);
+    if (import.meta.env.DEV) {
+      details = error.message;
+      stack = error.stack;
+    }
+  }
+
+  return (
+    <main>
+      <h1>{message}</h1>
+      <p>{details}</p>
+      {stack && (
+        <pre>
+          <code>{stack}</code>
+        </pre>
+      )}
+    </main>
+  );
+}`;
+
+const getServerSetupSnippet = (params: Params) => `
+import * as Sentry from "@sentry/react-router";${
+  params.isProfilingSelected
+    ? `
+import { nodeProfilingIntegration } from '@sentry/profiling-node';`
+    : ''
+}
+
+Sentry.init({
+  dsn: "${params.dsn.public}",
+
+  // Adds request headers and IP for users, for more info visit:
+  // https://docs.sentry.io/platforms/javascript/guides/react-router/configuration/options/#sendDefaultPii
+  sendDefaultPii: true,${
+    params.isProfilingSelected
+      ? `
+
+  integrations: [nodeProfilingIntegration()],`
+      : ''
+  }${
+    params.isPerformanceSelected
+      ? `
+  tracesSampleRate: 1.0, // Capture 100% of the transactions`
+      : ''
+  }${
+    params.isProfilingSelected
+      ? `
+  profilesSampleRate: 1.0, // profile every transaction`
+      : ''
+  }
+});`;
+
+const getServerEntrySnippet = () => `
+import * as Sentry from '@sentry/react-router';
+import { createReadableStreamFromReadable } from '@react-router/node';
+import { renderToPipeableStream } from 'react-dom/server';
+import { ServerRouter } from 'react-router';
+import { type HandleErrorFunction } from 'react-router';
+
+const handleRequest = Sentry.createSentryHandleRequest({
+  ServerRouter,
+  renderToPipeableStream,
+  createReadableStreamFromReadable,
+});
+
+export default handleRequest;
+
+export const handleError: HandleErrorFunction = (error, { request }) => {
+  // React Router may abort some interrupted requests, don't log those
+  if (!request.signal.aborted) {
+    Sentry.captureException(error);
+    // optionally log the error so you can see it
+    console.error(error);
+  }
+};`;
+
+const getVerifySnippet = () => `
+import type { Route } from "./+types/error-page";
+
+export async function loader() {
+  throw new Error("Sentry Test Error");
+}
+
+export default function ErrorPage() {
+  return <div>This page will throw an error!</div>;
+}`;
+
+const getPackageJsonScriptsSnippet = () => `
+{
+    "scripts": {
+      "dev": "NODE_OPTIONS='--import ./instrument.server.mjs' react-router dev",
+      "start": "NODE_OPTIONS='--import ./instrument.server.mjs' react-router-serve ./build/server/index.js"
+    }
+}`;
```

Comment:
Inconsistent indentation
```suggestion
const getPackageJsonScriptsSnippet = () => `
{
  "scripts": {
    "dev": "NODE_OPTIONS='--import ./instrument.server.mjs' react-router dev",
    "start": "NODE_OPTIONS='--import ./instrument.server.mjs' react-router-serve ./build/server/index.js"
  }
}`;
```